Added new application lms with basic functionality to implement new design.

alexbojko opened the pull request:
Added new application lms, templates for this app.

Commit:
feature(lms) Adds lms application.

Changes:
Adds url, view and basic templates for course page with list of curslets.
Test that course_view returns correct tontext to the template and that all needed functions were called during execution of this view.

New files:
- lms/
- lms/models.py
- lms/views.py
- lms/urls.py
- lms/tests.py
- lms/migrations
- lms/admin.py

cmltaWt0 reviewed (dismissed):
Review

cmltaWt0 commented on mysite/lms/views.py:
```diff
@@ -0,0 +1,22 @@
+
+
+@login_required
+def course_view(request, course_id):
```
Rewrite with ClassBasedView

cmltaWt0 commented on mysite/lms/views.py:
```diff
@@ -0,0 +1,22 @@
+# Create your views here.
+from ct.forms import NewUnitTitleForm, ReorderForm, set_crispy_action
+from ct.models import Course
+from ct.templatetags.ct_extras import is_teacher_url, md2html
```
There are a lot of unused imports.

cmltaWt0 commented on mysite/templates/lms/base.html:
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
Template should be rewrited.

cmltaWt0 commented on mysite/templates/lms/base.html:
```diff
@@ -0,0 +1,27 @@
+    {% include "google-analytics.html" %}
+  </head>
+  <body>
+    {% block content %}{% endblock %}
```
Wrong blocks structure.

cmltaWt0 commented on mysite/templates/lms/course_page.html:
```diff
@@ -0,0 +1,76 @@
+{% endif %}
+
+<ul class="nav nav-tabs">
+  {% for tabLabel,tabURL in pageData.navTabs %}
```
We do not need a legacy code here.

cmltaWt0 commented on mysite/templates/lms/course_page.html:
```diff
@@ -0,0 +1,76 @@
+{% if pageData.headText %}
+<input type="checkbox" id="headtoggle"/>Show {{ pageData.headLabel }}<BR>
+<div id="headdiv" style="display: none">
+{{ pageData.headText }}
```
pageData is absence.

cmltaWt0 commented on mysite/templates/lms/base.html:
```diff
@@ -0,0 +1,14 @@
+    {% block head %}{% endblock %}
+    {% include "google-analytics.html" %}
+  </head>
+  <title>{% block titile %}{% endblock %}</title>
```
typo

Commit:
feature(lms): Adds fsm plugin into lms application. Fixes in templates and tests.

Adds fsm_plugin into lms application with empty live_session fsm.
Fixed a typo in lms/base.html

New files:
 - lms/fsm_plugin/__init__.py
 - lms/fsm_plugin/live_session.py

cmltaWt0 commented:
```
./manage.py test lms
TypeError: 'Mock' object is not iterable
```

Commit:
feature(lms): Fix in course_view test, added negative test, fix in template.

Added negative test for course_view page, fix in template so now button Join will not be shown if there's no opened live session for course.
